app/models/concerns/host_info_extensions.rb

```diff
@@ -0,0 +1,40 @@
+# This module provides a method that exposes parameter information for a given host.
+# The hash is extensible, so any plugin/module can add a provider that will add its
+# information to the method.
+# Usage:
+# host = Host::Managed.find(my_host_id)
+# host.info
+# => A hash of hashes that describes all parameters associated with the host.
+#
+# Writing an extension:
+# Inherit HostInfo::Provider and override #host_info method and register the
+# provider by calling HostInfo.register_info_provider
+module HostInfoExtensions
+  extend ActiveSupport::Concern
+
+  included do
+    # Add default providers
+    HostInfo.register_info_provider(HostInfoProviders::StaticInfo)
+    HostInfo.register_info_provider(HostInfoProviders::ConfigGroupsInfo)
+    HostInfo.register_info_provider(HostInfoProviders::PuppetInfo)
+    HostInfo.register_info_provider(HostInfoProviders::HostParamsInfo)
+  end
+
+  # This is the base method that gathers information from all providers.
+  def info
+    renderer_regex = /renderer\.rb.*host_enc/
+    unless caller.first.match(renderer_regex) || caller[1].match(renderer_regex)
+      Foreman::Deprecation.renderer_deprecation('1.17', __method__, 'host_enc')
```

Comment:
My apologies I messed up, it should have remained 1.17. The deprecation might be removed completely soon but we should probably keep it same in this PR.